### ISSUE TITLE
Possible to apply the same tag twice / or / sometimes the same tag is displayed twice.

### ISSUE THREAD

**Issue by artfulrobot** [nextcloud/deck#2826]
### Steps to reproduce
I don't know, but here's proof it has happened:

![image](https://user-images.githubusercontent.com/870343/108966490-24c46000-7676-11eb-832e-47f735de7001.png)

Note ➊ shows "Wait on clients" tag twice, and that ➋ shows +2 tags as well as the shown "Wait on clients".

Nb. this is not a duplicate of #586 which is about allowing users to create two tags with identical names. I only have one tag with this name, but it's somehow showing twice.

Nb. I also use the Android deck app.

### Expected behaviour
It should not be possible to add the same tag twice.

### Actual behaviour
Somehow I've ended up with the same tag being applied twice.

## Server configuration detail

**Operating system:** Linux 4.19.0-9-amd64 #1 SMP Debian 4.19.118-2 (2020-04-29) x86_64

**Webserver:** nginx/1.14.2 (fpm-fcgi)

**Database:** mysql 10.3.27

**PHP version:** 

7.3.27-1~deb10u1
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, sodium, session, standard, cgi-fcgi, mysqlnd, PDO, xml, bcmath, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, gmp, iconv, igbinary, imagick, intl, json, exif, mysqli, pdo_mysql, Phar, posix, readline, redis, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Zend OPcache, xdebug

**Nextcloud version:** 20.0.6 - 20.0.6.1

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** unknown

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.6.0
 - activity: 2.13.4
 - admin_audit: 1.10.0
 - appointments: 1.8.1
 - apporder: 0.11.0
 - bookmarks: 4.0.8
 - bruteforcesettings: 2.0.1
 - calendar: 2.1.3
 - cloud_federation_api: 1.3.0
 - comments: 1.10.0
 - contacts: 3.4.3
 - contactsinteraction: 1.1.0
 - cookbook: 0.7.10
 - dashboard: 7.0.0
 - dav: 1.16.2
 - deck: 1.2.4
 - duplicatefinder: 0.0.4
 - federatedfilesharing: 1.10.2
 - federation: 1.10.1
 - files: 1.15.0
 - files_external: 1.11.1
 - files_fulltextsearch: 20.0.0
 - files_linkeditor: 1.1.3
 - files_pdfviewer: 2.0.1
 - files_rightclick: 0.17.0
 - files_sharing: 1.12.2
 - files_trashbin: 1.10.1
 - files_versions: 1.13.0
 - files_videoplayer: 1.9.0
 - firstrunwizard: 2.9.0
 - fulltextsearch: 20.0.0
 - fulltextsearch_elasticsearch: 20.0.0
 - impersonate: 1.7.0
 - integration_github: 0.0.17
 - issuetemplate: 0.7.0
 - logreader: 2.5.0
 - lookup_server_connector: 1.8.0
 - news: 15.2.2
 - nextcloud_announcements: 1.9.0
 - notes: 4.0.2
 - notifications: 2.8.0
 - oauth2: 1.8.0
 - password_policy: 1.10.1
 - passwords: 2020.12.2
 - photos: 1.2.3
 - polls: 1.7.5
 - previewgenerator: 3.1.1
 - privacy: 1.4.0
 - provisioning_api: 1.10.0
 - ransomware_protection: 1.8.0
 - recommendations: 0.8.0
 - serverinfo: 1.10.0
 - settings: 1.2.0
 - spreed: 10.0.5
 - systemtags: 1.10.0
 - tasks: 0.13.6
 - text: 3.1.0
 - theming: 1.11.0
 - theming_customcss: 1.8.0
 - twofactor_backupcodes: 1.9.0
 - updatenotification: 1.10.0
 - user_status: 1.0.1
 - viewer: 1.4.0
 - weather_status: 1.0.0
 - workflowengine: 2.2.0
Disabled:
 - encryption
 - sharebymail
 - support
 - survey_client
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "safe.artfulrobot.uk"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "overwrite.cli.url": "https:\/\/safe.artfulrobot.uk",
    "dbtype": "mysql",
    "version": "20.0.6.1",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "htaccess.RewriteBase": "\/",
    "overwritewebroot": "\/",
    "theme": "",
    "loglevel": 0,
    "maintenance": false,
    "updater.release.channel": "stable",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpmode": "smtp",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "simpleSignUpLink.shown": false,
    "mail_sendmailmode": "smtp",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "memcache.local": "\\OC\\Memcache\\Redis",
    "memcache.distributed": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 0,
        "dbindex": 0,
        "password": "***REMOVED SENSITIVE VALUE***",
        "timeout": 1.5
    },
    "app_install_overwrite": [
        "fulltextsearch_elasticsearch"
    ],
    "trashbin_retention_obligation": "365, auto"
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** 

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

## Client configuration

**Browser:** Mozilla/5.0 (X11; Linux x86_64; rv:86.0) Gecko/20100101 Firefox/86.0

**Comment by stefan-niedermann** [comment]
@juliushaertl There should be a unique constraint (https://github.com/nextcloud/deck/issues/2173) - Don't know whether you want to keep *this* issue open to find the actual root cause or close it as duplicate because the effects mentioned in this issue won't happen when #2173 has been implemented.

**Comment by vitis586** [comment]
I also just found out about this issue and I replicated it on current Opera, Chrome and Firefox versions. It behaves the same on my testing and production instance of current Deck and Nextcloud versions.
**Steps to reproduce**
1. Go to card
2. Fill in a new tag name and confirm it (either by hitting enter or by clicking on the plus) - this creates the new tag
3. You see that the tag is not assigned, so you go and assign it - this visibly assigns the tag to the card
4. Hit refresh and see the same tag assigned twice to the card

- The tag gets also assigned in the step 3, but that is only visible as a tick in the tag dropdown menu. The tag is not visible in the input field or on the board until you refresh the page with the board.
- You can close the card and reopen it before the step 3

**Removing the duplicate tag is also kind of annoying**
1. You spot the duplicate tag
2. Remove the duplicate tag - only one tag remains visible
3. Refresh page - both tags are gone
4. Add tag again - one tag visible, one tag assigned

Browser console does not show anything.

**Comment by manicmarvin** [comment]
Just want to bump this thread. This is a bug that's annoying/confusing/frustrating people in our organisation and I wondered if this will be solved soon?
